### PR TITLE
PYIC-1151: Dedupe sharedClaims in CredentialIssuerStartHandler response JAR.

### PR DESCRIPTION
**Pull request by Danomanic**
### What changed
Deduplicate sharedClaims in the JAR response.

### Issue tracking

- [PYIC-1151](https://govukverify.atlassian.net/browse/PYIC-1151)

### Environment variables or secrets

- [X] No environment variables or secrets were added or changed

### CODE REVIEW

File: lambdas/credentialissuerstart/src/test/java/uk/gov/di/ipv/core/credentialissuer/CredentialIssuerStartHandlerTest.java [hunk 1]
```diff
@@ -233,12 +233,45 @@ private void assertSharedClaimsJWTIsValid(String request)
...
+
+        SignedJWT signedJWT = SignedJWT.parse(jweObject.getPayload().toString());
+        JsonNode claimsSet = objectMapper.readTree(signedJWT.getJWTClaimsSet().toString());
+
+        JsonNode sharedClaims = claimsSet.get(SHARED_CLAIMS);
+        assertEquals(1, sharedClaims.get("name").size());
+        assertEquals(2, sharedClaims.get("birthDate").size());
+        assertEquals(2, sharedClaims.get("address").size());
+    }
+
```
Comment (blakeyp) [inline lambdas/credentialissuerstart/src/test/java/uk/gov/di/ipv/core/credentialissuer/CredentialIssuerStartHandlerTest.java:275]: Is it worth adding a test for the edge case in the ticket, where eg some parts of the name are the same but not all?

Comment (Danomanic) [reply lambdas/credentialissuerstart/src/test/java/uk/gov/di/ipv/core/credentialissuer/CredentialIssuerStartHandlerTest.java:275]: Patrick has added a test to confirm this edge case is dealt with by using `@EqualsAndHashCode`

File: lambdas/credentialissuerstart/src/test/java/uk/gov/di/ipv/core/credentialissuer/CredentialIssuerStartHandlerTest.java [hunk 2]
```diff
@@ -53,12 +53,7 @@
 import static org.mockito.Mockito.verifyNoInteractions;
 import static org.mockito.Mockito.when;
 import static uk.gov.di.ipv.core.credentialissuer.CredentialIssuerStartHandler.SHARED_CLAIMS;
-import static uk.gov.di.ipv.core.library.fixtures.TestFixtures.CREDENTIAL_ATTRIBUTES_1;
-import static uk.gov.di.ipv.core.library.fixtures.TestFixtures.CREDENTIAL_ATTRIBUTES_2;
-import static uk.gov.di.ipv.core.library.fixtures.TestFixtures.EC_PRIVATE_KEY;
-import static uk.gov.di.ipv.core.library.fixtures.TestFixtures.EC_PUBLIC_JWK;
-import static uk.gov.di.ipv.core.library.fixtures.TestFixtures.RSA_ENCRYPTION_PRIVATE_KEY;
-import static uk.gov.di.ipv.core.library.fixtures.TestFixtures.RSA_ENCRYPTION_PUBLIC_JWK;
+import static uk.gov.di.ipv.core.library.fixtures.TestFixtures.*;
```
Comment (Wynndow) [inline lambdas/credentialissuerstart/src/test/java/uk/gov/di/ipv/core/credentialissuer/CredentialIssuerStartHandlerTest.java]: Could this get switched back to the individual imports?
I'm wondering if we can get spotless to pick this up...

Comment (Danomanic) [reply lambdas/credentialissuerstart/src/test/java/uk/gov/di/ipv/core/credentialissuer/CredentialIssuerStartHandlerTest.java]: Yep, will sort this. Think Patrick's IntelliJ needs configuring for this.

Comment (Danomanic) [reply lambdas/credentialissuerstart/src/test/java/uk/gov/di/ipv/core/credentialissuer/CredentialIssuerStartHandlerTest.java]: Fixed.

**Comment by Wynndow** [comment]
~I'm getting a very weird error when trying to run your new tests locally. An aspectJ error, seems to be related to the X-ray tracing annotations. Did you run into anything like this?~
It was IntelliJ being IntelliJ. Ignore me.

File: lib/src/main/java/uk/gov/di/ipv/core/library/domain/Address.java [hunk 3]
```diff
@@ -2,13 +2,16 @@
 
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import com.fasterxml.jackson.annotation.JsonInclude;
+import lombok.EqualsAndHashCode;
 import uk.gov.di.ipv.core.library.annotations.ExcludeFromGeneratedCoverageReport;
 
 @JsonInclude(JsonInclude.Include.NON_NULL)
 @JsonIgnoreProperties(ignoreUnknown = true)
 @ExcludeFromGeneratedCoverageReport
+@EqualsAndHashCode
```
Comment (Wynndow) [inline lib/src/main/java/uk/gov/di/ipv/core/library/domain/Address.java:11]: This is pretty nifty. 

Why do we need to define them for all of these domain objects though?

Comment (Wynndow) [reply lib/src/main/java/uk/gov/di/ipv/core/library/domain/Address.java:11]: Is it the defining of these methods that is actually fixing the bug? So that the entries in the sets can be compared correctly?

Comment (Danomanic) [reply lib/src/main/java/uk/gov/di/ipv/core/library/domain/Address.java:11]: @Wynndow - We needed to add this, as when we add the name, address, dob to the set it was adding duplicates.

Java was using the default equals and hashcode method which was not comparing the fields within the objects - just that they matched in memory.

>  Its default implementation simply checks the object references of two objects to verify their equality. By default, two objects are equal if and only if they are refer to the same memory location.

By overriding equals and hashcode, we can ensure just the fields are compared. This had to be done for Name, NameParts, Address, BirthDate domain objects.

**Review by Wynndow — APPROVED** [review]
